Orange/widgets/visualize/owscatterplot.py

```diff
@@ -407,6 +410,38 @@ def handleNewSignals(self):
         self.apply_selection()
         self.unconditional_commit()
 
+    def prepare_data(self):
+        self.graph.new_data(self.sparse_to_dense(self.data_metas_X),
+                            self.sparse_to_dense(self.subset_data),
+                            new=False
+                           )
+        self.update_graph()
+
+    def sparse_to_dense(self, input_data=None):
+        if input_data is None:
+            return None
+        if not input_data.is_sparse():
+            return input_data
+        self.vizrank_button.setVisible(False)  # not for subset data?
```

Comment:
Also, once this gets disabled it is not re-enabled if dense data comes in. I suggest we add something like:
```
self.vizrank_button.setEnabled(not (self.data and self.data.is_sparse()))
```